word_segmentation.py

```diff
@@ -707,6 +728,220 @@ def perfom_bayesian_optimization(hunits_lower, hunits_upper, embedding_dim_lower
+        thrsh: shows what percent of the text we want to cover
+    """
+    cum_sum = 0
+    cnt = 0
+    for val in ratios.values():
+        cum_sum += val
+        cnt += 1
+        if cum_sum > thrsh:
+            break
+    print("number of different grapheme clusters = {}".format(len(ratios.keys())))
+    print("{} grapheme clusters form {} of the text".format(cnt, thrsh))
+    # plt.hist(ratios.values(), bins=50)
+    # plt.show()
+
+
+################################ Classes ################################
+
+class Line:
+    """
+    A class that stores different verions of a line: unsegmented, ICU segmented, and manually segmented (if exists).
+    Args:
+        unsegmented: the unsegmented version of the line
+        icu_segmented: the ICU segmented version of the line
+        man_segmented: the manually segmented version of the line (doesn't always exist)
+        icu_word_brkpoints: a list of word break points computed by ICU
+        man_word_brkpoints: a list of word break points computed manually (doesn't always exist)
+        char_brkpoints: a list of exgtended grapheme cluster breakpoints for the line
+    """
+    def __init__(self, input_line, input_type):
+        """
+        The __init__ function creates a new instance of the class based on the input line and its type.
+        Args:
+            input_line: the input line. It should be a clean line in one language.
+            input_type: determines what is the type of the input line. It can be segmented, icu_segmented, or
+            man_segmented.
+        """
+        if input_type == "unsegmented":
+            self.unsegmented = input_line
+            self.compute_char_brkpoints()
+            self.compute_icu_segmented()
+            self.icu_word_brkpoints = self.compute_word_brkpoints(self.icu_segmented)
+            self.man_segmented = None
+            self.man_word_brkpoints = None
+
+        if input_type == "icu_segmented":
+            self.icu_segmented = input_line
+            self.icu_word_brkpoints = self.compute_word_brkpoints(self.icu_segmented)
+            self.unsegmented = input_line.replace("|", "")
+            self.compute_char_brkpoints()
+            self.man_segmented = None
+            self.man_word_brkpoints = None
+
+        if input_type == "man_segmented":
+            self.man_segmented = input_line
+            self.man_word_brkpoints = self.compute_word_brkpoints(self.man_segmented)
+            self.unsegmented = input_line.replace("|", "")
+            self.compute_char_brkpoints()
+            self.compute_icu_segmented()
+            self.icu_word_brkpoints = self.compute_word_brkpoints(self.icu_segmented)
+
+        chars_break_iterator = BreakIterator.createCharacterInstance(Locale.getRoot())
+        chars_break_iterator.setText(self.unsegmented)
+        self.char_brkpoints = [0]
+        for brkpoint in chars_break_iterator:
+            self.char_brkpoints.append(brkpoint)
+
+    def compute_icu_segmented(self):
```

Comment:
Here and below: This function looks private.  The naming convention in the Google style guide is to prefix private methods with `_`, like `def _compute_icu_segmented(self):`